Standards/scs-0100-v4-flavor-naming.md

```diff
@@ -0,0 +1,549 @@
+
+#### Generation and Vendor
+
+The options for arch are as follows:
+
+| Letter  | vendor/architecture  | Corresponding image architecture  |
+| ------- | -------------------- | --------------------------------- |
+| (none)  | Generic x86-64       | `x86_64`                          |
+| `i`     | Intel x86-64         | `x86_64`                          |
+| `z`     | AMD (Zen) x86-64     | `x86_64`                          |
+| `a`     | ARM v8+              | `aarch64`                         |
+| `r`     | RISC-V               | (not yet listed in Glance)        |
+
+The generation is vendor specific and can be left out, but it can only be specified in
+conjunction with a vendor. At present, these values are possible:
+
+| Generation | i (Intel x86-64) | z (AMD x86-64) |  a (AArch64)       | r (RISC-V) |
+| ---------- | ---------------- | -------------- | ------------------ | ---------- |
+| 0          | pre Skylake      | pre Zen        | pre Cortex A76     | TBD        |
+| 1          | Skylake          | Zen-1 (Naples) | A76/NeoN1 class    | TBD        |
+| 2          | Cascade Lake     | Zen-2 (Rome)   | A78/x1/NeoV1 class | TBD        |
+| 3          | Ice Lake         | Zen-3 (Milan)  | A71x/NeoN2 (ARMv9) | TBD        |
+| 4          | Sapphire Rapids  | Zen-4 (Genoa)  |                    | TBD        |
+
+It is recommended to leave out the `0` when specifying the old generation; this will
+help the parser tool, which assumes 0 for an unspecified value and does leave it
+out when generating the name for comparison. In other words: 0 has a meaning of
+"rather old or unspecified".
+
+:::note
+
+We don't differentiate between Zen-4 (Genoa) and Zen-4c (Bergamo); L3 cache per
+Siena core is smaller on Bergamo and the frequency lower but the cores are otherwise
+identical. As we already have a qualifier `h` that allows to specify higher frequencies
+(which Genoa thus may use more and Bergamo less or not), we have enough distinction
+capabilities.
+
+:::
+
+#### Frequency Suffixes
+
+| Suffix | Meaning           |
+| ------ | ----------------- |
+| h      | >2.75GHz all-core |
+| hh     | >3.25GHz all-core |
+| hhh    | >3.75GHz all-core |
+
+#### Examples
+
+- SCS-2C-4-10n
+- SCS-2C-4-10n\_**z**
+- SCS-2C-4-10n\_**z3**
+- SCS-2C-4-10n\_**z3h**
+- SCS-2C-4-10n\_**z3hh**
+- SCS-2C-4-10n\_bms\_**z**
+- SCS-2C-4-10n\_bms\_**z3**
+- SCS-2C-4-10n\_bms\_**z3**
+- SCS-2C-4-10n\_bms\_**z3h**
+- SCS-2C-4-10n\_bms\_**z3hh** <- Bare Metal, AMD Milan with > 3.25GHz all core freq
+
+### [OPTIONAL] GPU support
+
+Format: `_`\[`G/g`\]X\[N\]\[`-`M\]\[`h`\]
+
+This extension provides more details on the specific GPU:
+
+- pass-through (`G`) vs. virtual GPU (`g`)
+- vendor (X)
+- generation (N)
+- number (M) of processing units that are exposed (for pass-through) or assigned; see table below for vendor-specific terminology
+- high-performance indicator (`h`)
+
+Note that the vendor letter X is mandatory, generation and processing units are optional.
```

Comment:
Just for my understanding: judging from the format above, the `G/g` marker and the high-perf indicator `h` are also optional right?

So `SCS-16V-64-500s_N` would be a valid minimal example just stating the existence of an nVidia GPU, correct?

Reply:
I think even that section is so far unchanged compared to v3. As Patrick stated

> Note: The initial commit just added the flavor naming document in version 4.

(Which admits multiple readings, but my interpretation was that he merely copied the file with an increased version number.)

Reply:
I am aware. As stated this question is just for my understanding. The format syntax of the original standard suggests that the `G/g` might be optional but I can't find any examples or statements confirming this.
I wanted to make sure that *just in case this is unintended* we could address this potential inaccuracy when @cah-patrickthiem will be editing this section anyway.

Reply:
It's not optional AFAICT
https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/flavor-naming/flavor_names.py#L358

Reply:
Then I think the usage of square brackets in "\[`G/g`\]" of the format specification is wrong and the brackets should be removed.

We could adjust this in the course of this refinement PR while we're at it since the syntax and explanations will most likely need to be updated anyway.

Reply:
I don't think this format specification can be "wrong" it that sense, because it's not that precise. We would have to use some more precise syntax (such as EBNF or something). We might do that, but then we should change the standard throughout to use this syntax.

Reply:
> Just for my understanding: judging from the format above, the `G/g` marker and the high-perf indicator `h` are also optional right?
> 
> So `SCS-16V-64-500s_N` would be a valid minimal example just stating the existence of an nVidia GPU, correct?

So we break the compatibility with the old naming and make life harder for the parser?

Reply:
> Then I think the usage of square brackets in "[`G/g`]" of the format specification is wrong and the brackets should be removed.

True. Either `G` or `g` is needed to indicate a GPU and indicates PT or Virt